services/blockchain-app-registry/shared/utils/downloadUtils.js

```diff
@@ -28,16 +28,17 @@ const logger = Logger();
 const getHTTPProtocolByURL = (url) => url.startsWith('https') ? https : http;
 
 const downloadAndExtractTarball = (url, directoryPath) => new Promise((resolve, reject) => {
+	logger.info(`Downloading and extracting file from ${url} to ${directoryPath}.`);
 	getHTTPProtocolByURL(url).get(url, (response) => {
 		if (response.statusCode === 200) {
 			response.pipe(tar.extract({ cwd: directoryPath }));
 			response.on('error', async (err) => reject(new Error(err)));
 			response.on('end', async () => {
-				logger.info('File downloaded successfully');
-				resolve();
+				logger.info('File downloaded and extracted successfully.');
+				setTimeout(resolve, 1); // Since the promise resolves earlier than expected
```

Comment:
Could it be prone to race condition?

Reply:
JS is single threaded, I'm not sure if this comment is valid here? Could you please elaborate your statement?

Reply:
I mean we are delaying execution by 1ms, what if the server is under pressure and it is not enough?

Reply:
Increase the delay to 100ms. This should be ample.